### ISSUE TITLE
Weird problem, please help

### ISSUE THREAD

**Issue by milanmuz** [Klaus2m5/6502_EhBASIC_V2.22#5]
Hello, I've successfully booted EhBasic with my homemade computer, everything seems to work, but I am getting weird numbers instead of integers. What could be the problem? This is an output, processor is w65c02s:

6502 EhBASIC [C]old/[W]arm ?

Memory size ? 4096

Memory size ? $0800

009.24 Bytes free

Enhanced BASIC 2.22

Ready
10PRINT"HELLO"
20GOTO10
RUN
HELLO
HELLO
HELLO
HELLO

Break in line 04.0256
Ready
LIST

0.3872 PRINT"HELLO"
0.0064 GOTO10

Ready

**Comment by Klaus2m5** [comment]
Hi,

this sounds to me like a memory corruption bug with the floating point 
accumulator. The floating point accumulator is also used to convert 
binary numbers like the line number or the bytes free to decimal. I had 
a similar bug once where an interrupt driven system routine used the 
same location as the floating point accumulator. The default location of 
the FAC1 is 5 bytes from $AC.

cheers, Klaus

**Comment by milanmuz** [comment]
Thank you very much, it does make sense, but I don't have such a routines, it's very minimal system, just 4k RAM (from 0) and ROM (from C000, unchanged, works in emulator), I tried to single step and with clock, always same and consistent. I have no idea what could use those locations.

**Comment by Klaus2m5** [comment]
So you don't have any background activity in your system? Nothing else 
using zero page RAM or stack? Do you preserve all registers on I/O-calls 
in min_mon.asm? What modifications did you make to the original source code?

**Comment by milanmuz** [comment]
No, nothing, there is no anything else. I didn't make any changes, it's the same code I used for Arduino 6502 emulator, and it works fine there, I am using this version of code on number of different emulators and never had any problem, first time I tried it on real hardware, that happened. I didn't preserve registers, not sure I understand, can you, please, explain what to do?

**Comment by Klaus2m5** [comment]
I would assume that you at least need to change the I/O addresses in 
min_mon.asm. It is either this part:

; setup for the 6502 simulator environment

IO_AREA    = $F000        ; set I/O area for this monitor

ACIAsimwr    = IO_AREA+$01    ; simulated ACIA write port
ACIAsimrd    = IO_AREA+$04    ; simulated ACIA read port

or this:

; byte out to simulated ACIA

ACIAout
     STA    ACIAsimwr        ; save byte to simulated ACIA
     RTS

; byte in from simulated ACIA

ACIAin
     LDA    ACIAsimrd        ; get byte from simulated ACIA
     BEQ    LAB_nobyw        ; branch if no byte waiting

     SEC                ; flag byte received
     RTS

LAB_nobyw
     CLC                ; flag no byte received
no_load                ; empty load vector for EhBASIC
no_save                ; empty save vector for EhBASIC
     RTS

If you alter registers X,Y for ACIAin and A,X,Y for ACIAout you must 
save and restore them as EhBASIC expects them to not change.

**Comment by milanmuz** [comment]
I see. No, I am using Arduino Mega to read/write F001/F004, so, I thought there is no need for change. I planned to change those later.

**Comment by Klaus2m5** [comment]
Didn't you say that you have ROM from $C000? How do you read/write to an 
Arduino? That sounds to me like a can of worms and I am probably unable 
to help you with that.

**Comment by milanmuz** [comment]
Yes, but those two addresses are excluded, ROM is not that big, so in that area is min_mon part provided by Arduino (not real ROM, it's work in progress), and that part is working, since I have serial out, I can type, running basic programs, the only problem is with numbers. Thank you for your help.

**Comment by milanmuz** [comment]
I solved it! I was faulty RAM, I swapped the chips and it works! Thanks!

6502 EhBASIC [C]old/[W]arm ?

Memory size ? 2048

1279 Bytes free

Enhanced BASIC 2.22

Ready
10PRINT"HELLO"
20GOTO10
RUN
HELLO
HELLO
HELLO
HELLO
HELLO
HELLO

Break in line 10
Ready
LIST

10 PRINT"HELLO"
20 GOTO10

Ready
10FORI=0TO10
20PRINTI
30NEXT
RUN
 0
 1
 2
 3
 4
 5
 6
 7
 8
 9
 10

Ready
LIST

10 FORI=0TO10
20 PRINTI
30 NEXT

Ready
